core/src/main/scala/cats/collections/Predicate.scala

```diff
@@ -46,28 +60,62 @@ abstract class Predicate[-A] extends scala.Function1[A, Boolean] { self =>
   /**
    * Return the opposite predicate
    */
-  def negate: Predicate[A] = Predicate(a => !apply(a))
+  def negate: Predicate[A] = Predicate.wrap[A](p.andThen {
+    case e: Now[Boolean] => if (e.value) Eval.False else Eval.True
+    case e => e.flatMap { if (_) Eval.False else Eval.True }
+  })
 
   /**
    * Return the opposite predicate
    */
   def unary_!(): Predicate[A] = negate
+
+  /**
+   * compose with a function the predicate.
```

Comment:
```suggestion
   * compose a function with the predicate.
```
typo on my part, will fix promptly